### PR TITLE
Feature: add hubself cert manage mode that use bearer token to bootstrap yurthub agent

### PR DESCRIPTION
**Pull request by rambohe-ch**
yurthub is currently communicating with master using the kubelet component's certificate(/etc/kubernetes/kubelet.conf), and when the node certificate expires, yurthub will not work because kubelet cannot update the node certificate via yurthub.

To solve this case, add node certificate generation and cycle updating ability to yurthub, so yurthub without kubelet node certificate can communicate with master. and when node certificate expires, yurthub can update it by itself.

### CODE REVIEW

File: pkg/yurthub/certificate/hubself/cert_mgr.go [hunk 1]
```diff
@@ -0,0 +1,596 @@
+package hubself
```
Comment (Fei-Guo) [inline pkg/yurthub/certificate/hubself/cert_mgr.go:1]: Add file header.

**Comment by Fei-Guo** [comment]
> To solve this case, add node certificate generation and cycle updating ability to yurthub, so yurthub without kubelet node certificate can communicate with master. and when node certificate expires, yurthub can update it by itself.

Can you elaborate:
1) Where to get the `JoinToken` when starting the cert manager?
2) What is the workflow of handling node cert expiration with the new code? I assume the `(ycm *yurtHubCertManager) Update()` method will be called but I don't see the caller.

**Comment by charleszheng44** [comment]
Overall, the PR is good. I just have a question regarding the "jointoken." When the yurthub starts to run, the kubelet should have connected to the kube-apiserver already. Can we reuse the kubelet's bootstrap config instead of passing the "jointoken" through the command line?

**Review by charleszheng44 — COMMENTED** [review]
Overall, the PR is good. I just have a question regarding the "jointoken." When the yurthub starts to run, the kubelet should have connected to the kube-apiserver already. Can we reuse the kubelet's bootstrap config instead of passing the "jointoken" through the command line?

**Review by charleszheng44 — APPROVED** [review]
/lgtm

File: pkg/yurthub/certificate/hubself/cert_mgr.go (hunk 1 above)
Comment (rambohe-ch) [reply pkg/yurthub/certificate/hubself/cert_mgr.go:17]: fixed

**Comment by rambohe-ch** [comment]
> > To solve this case, add node certificate generation and cycle updating ability to yurthub, so yurthub without kubelet node certificate can communicate with master. and when node certificate expires, yurthub can update it by itself.
> 
> Can you elaborate:
> 
> 1. Where to get the `JoinToken` when starting the cert manager?
when transfer  k8s worker node into  openyurt worker node, currently yurtctl deploy a yurthub on the node. so bootstrap join token will be created by yurtctl before yurthub deployment. and we will add this implementation in another pull request.

> 2. What is the workflow of handling node cert expiration with the new code? I assume the `(ycm *yurtHubCertManager) Update()` method will be called but I don't see the caller.
yeah, the caller for cert update will be added in another pull request

**Comment by rambohe-ch** [comment]
> Overall, the PR is good. I just have a question regarding the "jointoken." When the yurthub starts to run, the kubelet should have connected to the kube-apiserver already. Can we reuse the kubelet's bootstrap config instead of passing the "jointoken" through the command line?

if no join token is set for yurthub when hubself cert mode is selected, orginal kubelet.conf also will be used.

**Pull request merged**